core/authentication_api.php

```diff
@@ -764,29 +783,29 @@ function auth_does_password_match( $p_user_id, $p_test_password ) {
  * @access public
  */
 function auth_process_plain_password( $p_password, $p_salt = null, $p_method = null ) {
-	$t_login_method = config_get( 'login_method' );
-	if( $p_method !== null ) {
-		$t_login_method = $p_method;
-	}
+	$t_login_method = $p_method !== null ? $p_method : config_get( 'login_method' );
 
 	switch( $t_login_method ) {
 		case CRYPT:
-
 			# a null salt is the same as no salt, which causes a salt to be generated
 			# otherwise, use the salt given
 			$t_processed_password = crypt( $p_password, $p_salt );
 			break;
 		case MD5:
 			$t_processed_password = md5( $p_password );
 			break;
+		case SAFE_HASH:
+			$t_processed_password = password_hash( $p_password, PASSWORD_DEFAULT );
```

Comment:
I'm not a fan of auto-rolling algorithm without a specific MantisBT upgrade.  But I'm OK with doing it later once the field size is updated to 255.  Even if we change the db schema in a new release, using PASSWORD_DEFAULT today would mean that an old MantisBT release deployed on a server with a new PHP version would break.

Ideally, I would rather if we can select a specific algorithm as they become available and as we know it is compatible with our schema.